### PR TITLE
Adding changes to support s390x

### PR DESCRIPTION
**Pull request by duane-ibm**
Added changes for building promu on s390x. Also modified the crossbuild.go to add s390x support.
Can someone please review the changes?

### CODE REVIEW

File: cmd/crossbuild.go [hunk 1]
```diff
@@ -51,6 +51,9 @@ var (
 		"openbsd/arm": {"openbsd/armv7"},
 		"netbsd/arm":  {"netbsd/armv6", "netbsd/armv7"},
 	}
+	defaultS390XPlatforms = []string{
```
Comment (SuperQ) [inline cmd/crossbuild.go]: Do we want to call this `defaultS390Platforms`, just in case we ever add support for both `s390` and `s390x`? Or is that silly?

**Comment by SuperQ** [comment]
Don't forget to update `doc/examples/.promu.yml`.

File: cmd/crossbuild.go (hunk 1 above)
Comment (SuperQ) [reply cmd/crossbuild.go]: No, it should be `defaultS390Platforms`. It also needs to be moved up to below `defaultMIPSPlatforms`.

**Comment by duane-ibm** [comment]
Done. Please review.

**Pull request merged**